src/Spout/Common/Helper/FileSystemHelper.php

```diff
@@ -117,12 +117,16 @@ public function deleteFolderRecursively($folderPath)
      * should occur is not inside the base folder.
      *
      * @param string $operationFolderPath The path of the folder where the I/O operation should occur
-     * @throws \Box\Spout\Common\Exception\IOException If the folder where the I/O operation should occur is not inside the base folder
+     * @throws \Box\Spout\Common\Exception\IOException If the folder where the I/O operation should occur
+     * is not inside the base folder or the base folder does not exist
      * @return void
      */
-    protected function throwIfOperationNotInBaseFolder($operationFolderPath)
+    protected function throwIfOperationNotInBaseFolder(string $operationFolderPath)
     {
         $operationFolderRealPath = \realpath($operationFolderPath);
+        if (!$this->baseFolderRealPath) {
```

Comment:
should this check be in the constructor instead? If `baseFolderRealPath` is `false`, it means that `realpath` call in the ctor failed; so we can catch the error there directly?

Reply:
This would also work with a small adjustments of the tests. The tests rely on a non existing base folder - maybe a user does, too? I could not think of an use case, though.

Reply:
Anyway, it's probably not worth making the change. The program fails pretty quickly with the current implementation.